src/winforms/toga_winforms/widgets/webview.py

```diff
@@ -96,6 +100,14 @@ def winforms_initialization_completed(self, sender, args):
             else:
                 print(args.InitializationException)
 
+    def winforms_web_resource_requested(self, sender, args):
+        if self.interface.on_webview_loading:
+            event_args = {}
+            event_args["request"] = args.Request
+            event_args["resource_context"] = args.ResourceContext
+            event_args["response"] = args.Response
+            self.interface.on_webview_loading(self.interface, event_args)
```

Comment:
I agree that the handler in the user code should not get platform specific data. But how can this be achieved? With a data class on the interface layer which is filled from the platform code? My problem here is that I don't know what event data other platforms deliver.

Reply:
As discussed on Discord, I wasn't suggesting you use `requests` *instead* of `WebView`, because I guess some user interaction with the web page is required. I was suggesting that the `requests` classes for request and response might be a good choice for a platform-neutral way to send that information to the handler, because many Python developers are already familiar with it.

If the other platforms support intercepting the request and response in a similar way, then it should be possible to populate at least the basic fields of those classes. But that should be verified before we add this to the public API.

Reply:
Yes, that is a good idea! I'll check if I can map the Winform event args onto those classes. Thanks for the suggestion

Reply:
@mhsmith: I looked at the Request class in requests and found that the headers attribute is a dictionary. As far as I know, this is incorrect, because a dict will not support having several entries with the same key. But in http, it *is* possible to have the same http header field several times in one request. We should therefore rather implement a list of tupples.

Reply:
According to https://github.com/psf/requests/issues/1415, `requests` returns repeated headers as a comma-separated list, which is consistent with the HTTP standard. It looks like this causes an issue with the Set-Cookie header, which can contain commas itself, but there's a workaround mentioned at https://github.com/psf/requests/issues/3957.

Reply:
The standard library [`HTTPResponse`](https://docs.python.org/3/library/http.client.html#httpresponse-objects) class is also an option.

Reply:
I don't intend to use third party classes or libraries. I just take them as inspiration for what attributes to include in the Request / Response classes that I intend to add to Toga's webview.py

Reply:
> FWIW: I'd prefer to avoid leaning on a third-party library like requests for functionality like this

Fair enough, but if the Python standard library has existing classes for HTTP requests and responses, then I recommend using those classes if possible rather than creating new ones.

Reply:
Looking at this closer - I'm not sure I see why the event is being pass in at all, or why a `request` object is required. The end user doesn't control the full HTTP request; they only have control over the URL that is going to be requested. Why not just pass the URL that is loading as an argument? What is the purpose of passing the rest of the request detail?

Reply:
Sure - an actual web request has lots of pieces that are configurable and might be of interest - but as an end user operating a browser, *none* of them are visible. The only user interface you have is with the top-level URL bar. You can choose a URL, and you wait for it to load. You can't set different request headers - you can't even change the request *method* - all browser requests initiated via the URL bar are GETs.

You only get control over the method and request header detail if you're dealing with an XMLHttpRequest - but that doesn't generate a load event in any browser I've seen.

So - I guess my actual questions is "What is the use case you're envisaging for accessing low-level request data as an API on a web browser control?"

Reply:
As I summarized [above](https://github.com/beeware/toga/pull/1508#discussion_r904942589), I think he's trying to intercept the response (not the request) to extract an access token, and then block the WebView from rendering it so he can use the token programmatically instead. @t-arn: is that correct?

Reply:
Ok.. so... I'm definitely not following. Do you want detail from the *request* or the *response*? And if it's the response... why would a "started load" event have access to *response* data? 

Can you give an example of the oAuth2 workflow that you're using? The examples of oAuth2 that I'm familiar with involve using callback URIs, not introspection of the *request* for a token.

I'll also note that it's [generally considered a bad idea to use embedded browsers for oAuth2 authentication](https://aaronparecki.com/oauth-2-simplified/#mobile-apps), because that doesn't give the user any way to validate whether the browser is phishing. My experience has been that you use an external app or the system browser for authentication, and return authentication responses via a custom URI handler. That's something Toga/Briefcase doesn't currently support, but authentication is clearly a use case for which we're going to need to add this support.